salt/grains/core.py

```diff
@@ -1390,7 +1396,8 @@ def os_data():
         shortname = distroname.replace(' ', '').lower()[:10]
         # this maps the long names from the /etc/DISTRO-release files to the
         # traditional short names that Salt has used.
-        grains['os'] = _OS_NAME_MAP.get(shortname, distroname)
+        if 'os' not in grains:
```

Comment:
@cachedout I understand your concern, but currently I think there is no problem here. Before this PR, `grains['os']` in **linux systems** is always picked from `_OS_NAME_MAP` and there isn't custom assignments for this grain in this scope.

Now, with the PR, the only case where `grains['os']` is set manually is our SUSE cases and we don't want to overwrite it later with the default value of `_OS_NAME_MAP`. So the default behavior for the `os` grain assignment will be the same that currently is except for SUSE systems.

What do you think?

Thanks @isbm for the comment